src/main/scala/net/snowflake/spark/snowflake/CloudCredentialsUtils.scala

```diff
@@ -19,26 +19,31 @@ package net.snowflake.spark.snowflake
 
 import java.net.URI
 
-import com.amazonaws.auth.{BasicAWSCredentials, AWSCredentials, AWSSessionCredentials, InstanceProfileCredentialsProvider}
+import com.amazonaws.auth.{AWSCredentials, AWSSessionCredentials, BasicAWSCredentials, InstanceProfileCredentialsProvider}
+import com.microsoft.azure.storage.StorageCredentialsSharedAccessSignature
 import net.snowflake.spark.snowflake.Parameters.MergedParameters
 import org.apache.hadoop.conf.Configuration
 import org.apache.spark.sql.SQLContext
 
-private[snowflake] object AWSCredentialsUtils {
+private[snowflake] object CloudCredentialsUtils {
   /**
    * Generates a credentials string for use in Snowflake COPY in-out statements.
    */
   def getSnowflakeCredentialsString(sqlContext: SQLContext,
-                                    params: MergedParameters): String = {
+                                          params: MergedParameters): String = {
     if (params.rootTempDir.startsWith("file://")) {
       "-- file URL, no creds needed"
+    } else if (params.rootTempDir.startsWith("wasb://") ||
```

Comment:
I wonder if we should add a lazy boolean in `MergedParameters` that will signal Azure vs. S3, i.e. something like `isAzure`. This would alleviate the need to do constant string checking. That method can later also be extended to resolve the stage-type for when `tempDir` isn't provided, i.e. when using the internal stage, as well as include other checks.

Reply:
I think it is cool but in that case I don't want to make it a boolean but a enum. Essentially, we have 3 types of filesystem here: local file, S3 and Azure. Also it will be a bit more extensible later. 
Let me make that change.